src/main/java/com/pidoco/juri/JURI.java

```diff
@@ -974,6 +995,70 @@ public JURI replaceQueryParametersMulti(Map<String, Collection<String>> params)
         return this;
     }
 
+    /**
+     * Navigate to a new URI.<br>
+     * The navigate method tries to mimic browser behaviour:
+     * 'What happens if you are on the current URI and click on the (relative or absolute) link.'<br>
+     * The method changes the current JURI inplace, it does not create a new object.
+     * If the path is altered, it is normalized.<br>
+     * <br>
+     * Examples:
+     * <ul>
+     *   <li>JURI.parse("http://example.com/a/b.html").navigate("http://www.google.com/search?q=2") =&gt; "http://www.google.com/search?q=2"</li>
+     *   <li>JURI.parse("http://example.com/a/b.html").navigate("c.html")                           =&gt; "http://example.com/a/c.html"</li>
+     *   <li>JURI.parse("http://example.com/a/b.html").navigate("c.html?z=1")                       =&gt; "http://example.com/a/c.html"</li>
+     *   <li>JURI.parse("http://example.com/a/b.html").navigate("../c.html")                        =&gt; "http://example.com/c.html"</li>
+     *   <li>JURI.parse("http://example.com/a/b.html").navigate("../../../../c.html")               =&gt; "http://example.com/c.html"</li>
+     *   <li>JURI.parse("http://example.com/a/b.html").navigate("#anchor")                          =&gt; "http://example.com/a/b.html#anchor"</li>
+     * </ul>
+     *
+     * @param path May be a relative path or a absolute URI to navigate to or an url fragment. 
+     * @return this.
+     */
+    public JURI navigate(String path) {
+        JURI newURI = JURI.parse(path);
+        if (newURI.getHost() != null) {
+            return this.resetBlank()
+                    .parseNew(path);
+        }
+
+        String newPath = newURI.getPath();
+        Map<String, Collection<String>> newQuery = newURI.getQueryParameters();
+        String newFragment = newURI.getFragment();
+
+        if (newURI.isHavingPath()) {
+            // set new path (and maybe query and fragment):
+            if (newURI.isPathRelative()) {
+                // relative path:
+                // calculate new normalized absolute path
+                newPath = this.clone()
+                        .addRawPath("../" + newURI.getPath())
+                        .getCurrentUri()
+                        .normalize()
+                        .getPath();
+                while (newPath.startsWith("/../")) {
```

Comment:
@tec Maybe add a comment on this line: ".. in root directory refers to the root directory."

Sorry this was something I still wanted to check and didn't have time when I looked at this last time.